fulfillment.go

```diff
@@ -43,22 +43,39 @@ type FulfillmentServiceOp struct {
-	OrderID         int64      `json:"order_id,omitempty"`
-	LocationID      int64      `json:"location_id,omitempty"`
-	Status          string     `json:"status,omitempty"`
-	CreatedAt       *time.Time `json:"created_at,omitempty"`
-	Service         string     `json:"service,omitempty"`
-	UpdatedAt       *time.Time `json:"updated_at,omitempty"`
-	TrackingCompany string     `json:"tracking_company,omitempty"`
-	ShipmentStatus  string     `json:"shipment_status,omitempty"`
-	TrackingNumber  string     `json:"tracking_number,omitempty"`
-	TrackingNumbers []string   `json:"tracking_numbers,omitempty"`
-	TrackingUrl     string     `json:"tracking_url,omitempty"`
-	TrackingUrls    []string   `json:"tracking_urls,omitempty"`
-	Receipt         Receipt    `json:"receipt,omitempty"`
-	LineItems       []LineItem `json:"line_items,omitempty"`
-	NotifyCustomer  bool       `json:"notify_customer"`
+	ID                          int64                        `json:"id,omitempty"`
+	OrderID                     int64                        `json:"order_id,omitempty"`
+	LocationID                  int64                        `json:"location_id,omitempty"`
+	Status                      string                       `json:"status,omitempty"`
+	CreatedAt                   *time.Time                   `json:"created_at,omitempty"`
+	Service                     string                       `json:"service,omitempty"`
+	UpdatedAt                   *time.Time                   `json:"updated_at,omitempty"`
+	TrackingCompany             string                       `json:"tracking_company,omitempty"`
+	ShipmentStatus              string                       `json:"shipment_status,omitempty"`
+	TrackingInfo                FulfillmentTrackingInfo      `json:"tracking_info,omitempty"`
+	TrackingNumber              string                       `json:"tracking_number,omitempty"`
+	TrackingNumbers             []string                     `json:"tracking_numbers,omitempty"`
+	TrackingUrl                 string                       `json:"tracking_url,omitempty"`
+	TrackingUrls                []string                     `json:"tracking_urls,omitempty"`
+	Receipt                     Receipt                      `json:"receipt,omitempty"`
+	LineItems                   []LineItem                   `json:"line_items,omitempty"`
+	LineItemsByFulfillmentOrder []LineItemByFulfillmentOrder `json:"line_items_by_fulfillment_order,omitempty"`
+	NotifyCustomer              bool                         `json:"notify_customer"`
+}
+
+// LineItemByFulfillmentOrder represents the FulfillmentOrders (and optionally the items) used to create a Fulfillment.
+// https://shopify.dev/docs/api/admin-rest/2023-01/resources/fulfillment#post-fulfillments
+type LineItemByFulfillmentOrder struct {
+	FulfillmentOrderID        int64      `json:"fulfillment_order_id,omitempty"`
+	FulfillmentOrderLineItems []LineItem `json:"fulfillment_order_line_items,omitempty"`
```

Comment:
`FulfillmentOrderLineItems []LineItem`
I don't think this is right, the docs say:
```
fulfillment_order_line_items: (array) The fulfillment order line items and 
the quantity of each which should be fulfilled. If this property is undefined,
then all of the fulfillment order line items of the associated fulfillment
order are fulfilled.
    id: (integer) (required) The ID of the fulfillment order line item.
    quantity: (integer) (required) (minimum: 1) The quantity of the fulfillment order line item.
```
And think you'll need another struct (or simply inline it).

Reply:
@AndrewChau - what do you think?

Reply:
Hi @oliver006 , I just opted to just use `LineItem` because it omits the empty fields but yes, it may confuse users. I found there is another existing struct called `FulfillmentOrderLineItemQuantity` which we can use.

Reply:
Let's keep it clean and not re-use structs that sort of resemble what we need.
Can you define a simple struct with "id" and "quantity" and use that here?

Reply:
👍